src/frameworks/gatsby5.json

```diff
@@ -0,0 +1,37 @@
+{
+  "id": "gatsby5",
+  "name": "Gatsby 5",
+  "category": "static_site_generator",
+  "detect": {
+    "npmDependencies": [{ "name": "gatsby", "version": ">=5.0.0" }],
+    "excludedNpmDependencies": [],
+    "configFiles": ["gatsby-config.js"]
+  },
+  "dev": {
+    "command": "gatsby develop",
+    "port": 8000,
+    "pollingStrategies": [{ "name": "TCP" }, { "name": "HTTP" }]
+  },
+  "build": {
+    "command": "gatsby build",
+    "directory": "public"
+  },
+  "staticAssetsDirectory": "static",
+  "env": {
+    "GATSBY_LOGGER": "yurnalist",
+    "GATSBY_PRECOMPILE_DEVELOP_FUNCTIONS": "true",
+    "AWS_LAMBDA_JS_RUNTIME": "nodejs18.x",
+    "NODE_VERSION": "18"
```

Comment:
These variables seem to only be used in the CLI and I'm not even certain about this. I haven't tested this because I think what we want cannot be achieved with setting NODE_VERSION. What we want is an explicit setting `minimumRequiredNodeVersion`.

Reply:
I'm not a fan of needing to create a new framework for each version. It feels hacky and hard to maintain. What happens when Gatsby 6 is released? Do we create a new one, or rename it to gatsby5plus? The approach of the plugins repo of allowing an array of versions, each with their own rules, seems to be better, though would be more of a breaking change.

Reply:
To answer your question - I think you are right, and these are only used in `ntl dev`, so `AWS_LAMBDA_JS_RUNTIME` is also incorrect here.

Reply:
I just followed the pattern that existed for other frameworks as everything else would be a big change.
I think that it wouldn't even be a breaking change, just a lot of internal changes. The returned result could stay the same.

What I'm thinking of is something like this (simplified)

```
{
  "id": "gatsby5",
  "name": "Gatsby",
  "detect": {
    "npmDependencies": ["gatsby"],
    "excludedNpmDependencies": [],
    "configFiles": ["gatsby-config.js"]
  },
  "overrides": [
    {
      "detect": {
        "npmDependencies": [{name: "gatsby", "version": ">99"],
      },
      "dev": {
        "port": 1111,
      },
    }
  ],
  "dev": {
    "port": 8888,
  },
}
```
So in this example it would change the dev port if a certain version of the framework is used, but inherit everything else from the initial config. The structure in the end result would be exactly the same.